### ISSUE TITLE
Cover output of spawned processes that inherit stdio

### ISSUE THREAD

**Issue by medikoo** [sindresorhus/hook-std#9]
It might be nice to somehow support that, at least via option.

Without that, output that hook-std captures doesn't necessary cover all output as seen from initialized process in a console.

I think it's possible only via overriding `ChildProcess.prototype.spawn` and act accordingly after recognizing the intention.

**Comment by sindresorhus** [comment]
I would like this too, but I don't think it's possible. From what I can remember, the stdio inheriting logic is done in native code in libuv, not in `ChildProcess.prototype.spawn`. But happy to be proven wrong in the form of a pull request 😀.

**Comment by medikoo** [comment]
@sindresorhus I investigated it and indeed it looks there's no way. Still I would leave it open, maybe other adventurers will find some miraculous solution.

**Comment by sindresorhus** [comment]
Sure

**Comment by orta** [comment]
As a passer-by who hit this issue, I don't know if future folks have the same kind of problems as me (I wanted guarantees that I was getting pretty much the exact stdout/stderr with ASNI, but not spamming the console.) 

What I did instead was use [node-pty](https://github.com/Microsoft/node-pty) to spawn an in-memory pseudo-terminal which I could extract _pretty much_  the exact same stdout/stderr.

**Comment by sindresorhus** [comment]
@orta Yup. That works, but `node-pty` is a native Node.js addon, which comes with all kinds of problems. It's not something I would depend on in a reusable package. If only Node.js had PTY built-in...

- https://github.com/nodejs/node/issues/31409
- https://github.com/nodejs/node-v0.x-archive/issues/2754
